From: Jesse Barnes <jbarnes@virtuousgeek.org>
Date: Tue Nov 12 10:17:39 2013 -0800
Subject: x86/early quirk: use gen6 stolen detection for VLV
Patch-mainline: Upstream/v3.13-rc1
Git-commit: 7bd40c16ccb2cb6877dd00b0e66249c171e6fa43

References: bnc#907970, FATE#316038

Signed-off-by: Egbert Eich <eich@suse.com>

We've always been able to use either method on VLV, but it appears more
recent BIOSes only support the gen6 method, so switch over to that.

References: https://bugs.freedesktop.org/show_bug.cgi?id=71370
Signed-off-by: Jesse Barnes <jbarnes@virtuousgeek.org>
Reviewed-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
---
 arch/x86/kernel/early-quirks.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- linux-3.12-SLE12.orig/arch/x86/kernel/early-quirks.c
+++ linux-3.12-SLE12/arch/x86/kernel/early-quirks.c
@@ -317,8 +317,8 @@ static struct pci_device_id intel_stolen
 	INTEL_I915GM_IDS(gen3_stolen_size),
 	INTEL_I945G_IDS(gen3_stolen_size),
 	INTEL_I945GM_IDS(gen3_stolen_size),
-	INTEL_VLV_M_IDS(gen3_stolen_size),
-	INTEL_VLV_D_IDS(gen3_stolen_size),
+	INTEL_VLV_M_IDS(gen6_stolen_size),
+	INTEL_VLV_D_IDS(gen6_stolen_size),
 	INTEL_PINEVIEW_IDS(gen3_stolen_size),
 	INTEL_I965G_IDS(gen3_stolen_size),
 	INTEL_G33_IDS(gen3_stolen_size),
